packages/@aws-cdk-testing/framework-integ/test/aws-apigateway/test/integ.restapi.vpc-endpoint.ts

```diff
@@ -1,49 +1,138 @@
+            statusCode: 200,
           }),
+        },
+        integrationResponses: [
+          {
+            statusCode: '200',
+            responseTemplates: {
+              'application/json': JSON.stringify({ message: 'OK' }),
+            },
+          },
         ],
       }),
+      {
+        methodResponses: [
+          {
+            statusCode: '200',
+          },
+        ],
+      },
+    );
+    this.api.grantInvoke(vpcEndpoint);
+
+    this.testFunction = new lambda.Function(this, 'TestFunction', {
+      runtime: lambda.Runtime.NODEJS_20_X,
+      handler: 'index.handler',
+      code: lambda.Code.fromInline(`
+        const https = require('https');
+
+        const httpsGet = (url) => {
+          return new Promise((resolve, reject) => {
+            https.get(url, (res) => {
+              let data = '';
+
+              res.on('data', (chunk) => {
+                data += chunk;
+              });
+
+              res.on('end', () => {
+                resolve(data);
+              });
+
+              res.on('error', (e) => {
+                reject(e);
+              });
+            });
+          });
+        };
+
+        exports.handler = async function(event) {
+          const apiEndpoint = process.env.API_ENDPOINT;
+
+          try {
+            const data = await httpsGet(apiEndpoint);
+            return {
+              statusCode: 200,
+              body: JSON.stringify({ message: 'Success', data }),
+            };
+          } catch (error) {
+            return {
+              statusCode: 500,
+              body: JSON.stringify({ message: 'Error', error: error.message }),
+            };
+          }
+        };
+      `),
+      vpc,
+      environment: {
+        API_ENDPOINT: this.api.url,
+      },
     });
-    api.root.addMethod('GET');
+    this.testFunction.addToRolePolicy(
+      new iam.PolicyStatement({
+        actions: ['execute-api:Invoke'],
+        resources: [this.api.arnForExecuteApi()],
+      }),
+    );
+    this.testFunction.connections.allowTo(vpcEndpoint, ec2.Port.tcp(443));
   }
 }
 
 const app = new cdk.App();
 
-const testCase = new Test(app, 'test-apigateway-vpcendpoint');
-new IntegTest(app, 'apigateway-vpcendpoint', {
+const testCase = new TestStack(app, 'ApiGatewayVpcEndpointTestStack');
+const integ = new IntegTest(app, 'ApiGatewayVpcEndpointInteg', {
   testCases: [testCase],
 });
 
-app.synth();
+const assertion = integ.assertions.awsApiCall('Lambda', 'invoke', {
+  FunctionName: testCase.testFunction.functionName,
+  InvocationType: 'RequestResponse',
+});
+assertion.provider.addToRolePolicy({
+  Effect: 'Allow',
+  Action: ['lambda:InvokeFunction'],
+  Resource: [testCase.testFunction.functionArn],
+});
+assertion.expect(
+  ExpectedResult.objectLike({
+    StatusCode: 200,
+    Payload: '{\'statusCode\':200,\'body\':\'{\\\'message\\\':\\\'Success\\\',\\\'data\\\':\\\'{\\\\\\\'message\\\\\\\':\\\\\\\'OK\\\\\\\'}\\\'}\'}',
+  }),
+);
```

Comment:
Nice!! I love seeing assertions in integ tests to test functionality instead of simply deployability.